pandas/tests/io/formats/test_to_excel.py

```diff
@@ -212,3 +214,50 @@ def test_css_to_excel_multiple():
 def test_css_to_excel_inherited(css, inherited, expected):
     convert = CSSToExcelConverter(inherited)
     assert expected == convert(css)
+
+
+@pytest.mark.parametrize("input_color,output_color", (
+    [(name, rgb) for name, rgb in CSSToExcelConverter.NAMED_COLORS.items()] +
+    [("#" + rgb, rgb) for rgb in CSSToExcelConverter.NAMED_COLORS.values()] +
+    [("#F0F", "FF00FF"), ("#ABC", "AABBCC")] +
+    [(None, None), ("not-a-color", None)])
+)
+def test_css_to_excel_colors(input_color, output_color):
+    # see gh-18392
+    css = ("border-top-color: {color}; "
+           "border-right-color: {color}; "
+           "border-bottom-color: {color}; "
+           "border-left-color: {color}; "
+           "background-color: {color}; "
+           "color: {color}").format(color=input_color)
+
+    expected = {}
+
+    if input_color is not None:
+        expected["fill"] = {
+            "patternType": "solid"
+        }
+
+        if output_color is not None:
+            expected["border"] = {
+                k: {
+                    "color": output_color,
+                } for k in ("top", "right", "bottom", "left")
+            }
+
+            expected["font"] = {
+                "color": output_color
+            }
+
+            expected["fill"]["fgColor"] = output_color
+
+    if input_color in (None, "not-a-color"):
```

Comment:
can you separate this into 2 tests rather than do the if (IOW good colors) and bad colors (which show a warning?)